src/backend/parser/gram.y

```diff
@@ -16451,7 +16451,7 @@ qualified_name:
 				{
 					$$ = makeRangeVar(NULL, $1, @1);
 					/* TSQL temp table names */
-					if (strncmp($1, "#", 1) == 0 || strncmp($1, "##", 2) == 0)
+					if (sql_dialect == SQL_DIALECT_TSQL && strncmp($1, "#", 1) == 0 || strncmp($1, "##", 2) == 0)
```

Comment:
Please add the OR BLOCK inside it's own brackets

Reply:
Done